src/olympia/bandwagon/serializers.py

```diff
@@ -66,18 +68,35 @@ def validate_slug(self, value):
 
         return value
 
-    def validate_author(self, value):
-        if not self.partial:
-            # If we've got a new collection set the author to account user
-            value = self.context['request'].user
-        # Otherwise, we're modifying an existing collection so don't change it.
-        return value
+
+class ThisCollectionDefault(object):
+    def set_context(self, serializer_field):
+        viewset = serializer_field.context['view']
+        self.collection = viewset.get_collection_viewset().get_object()
+
+    def __call__(self):
+        return self.collection
 
 
 class CollectionAddonSerializer(serializers.ModelSerializer):
-    addon = AddonSerializer()
-    notes = TranslationSerializerField(source='comments')
+    addon = SplitField(
+        SlugOrPrimaryKeyRelatedField(
+            queryset=Addon.objects.public().no_cache()),
```

Comment:
Something like that, yet. Do add that it should not be a big issue because that's only used for writes, not reads.